Refactor: Conduit Permissions

amusingaxl opened the pull request:
Granting a role to `address(0)` makes the methods exclusive to that role permissionless.

This is an alternative to #25.

amusingaxl committed:
refactor: allow conduits to have permissionless methods

Granting a role to `address(0)` makes the methods exclusive to that role permissionless.

dbkbali reviewed (commented):
Good work on this @amusingaxl .. thanks for taking the time to review this with me. My personal view is that if the auth modifer change is likely to never be used is to leave this change out - so create a new set of conduits with just the ability to set to make the operator/mate roles permissioless

amusingaxl committed:
refactor: remove `address(0)` convention for `wards`

This feature will most likely never be used, so we will add this only for operators and mates.

amusingaxl commented:
Closing this in favor of #25
